src/app/public/modules/wait/wait-adapter.service.ts

```diff
@@ -1,13 +1,28 @@
 import {
   ElementRef,
   Injectable,
-  Renderer
+  Renderer,
+  OnDestroy
 } from '@angular/core';
 
+import {
+  SkyWaitComponent
+} from './wait.component';
+
 @Injectable()
-export class SkyWaitAdapterService {
+export class SkyWaitAdapterService implements OnDestroy {
+  private parentListeners: {[key: string]: Function} = {};
 
-  constructor(private renderer: Renderer) { }
+  constructor(
+    private renderer: Renderer
+  ) { }
+
+  public ngOnDestroy() {
+    for (let key of Object.keys(this.parentListeners)) {
+      this.parentListeners[key]();
```

Comment:
Why this line (22)? I only expected to find just the delete here.

Reply:
Invoking the method stops the listener from running, then on the following line I delete my reference to it